tensorflow_probability/examples/jupyter_notebooks/Linear_Mixed_Effects_Models.ipynb

```diff
@@ -590,11 +588,17 @@
     "    ])\n",
     "\n",
     "    # To enable tracking of the trainable variables via the created distribution,\n",
-    "    # we attach a reference to `self`. Since all TFP objects sub-class\n",
-    "    # `tf.Module`, this means that the following is possible:\n",
-    "    # LinearMixedEffectModel()(features_train).trainable_variables\n",
-    "    # ==> tuple of all tf.Variables created by LinearMixedEffectModel.\n",
-    "    model._to_track = self\n",
+    "    # we attach a reference to `self`. Since all TFP objects (e.g. `JointDistributionSequential` in this case )\n",
+    "    # sub-class `tf.Module`, this means that if we attach an object of \n",
+    "    # type tf.Module (i.e. `LinearMixedEffectModel` in this case) to the return value then the variables created\n",
+    "    # by `LinearMixedEffectModel` will be automatically tracked\n",
+    "    #        \n",
+    "    # In the below statement, we have used `lmm_model` as the name of the property but we could have\n",
```

Comment:
I still think that including the example that was here previously is helpful, and can be called back to in the second comment you added.

I would also remove the part of this comment after "In the below statement" -- I think your renaming makes it clear this is not a private flag, and you already mentioned that `tf.Module` tracks trainable variables above.

Reply:
If you come from other languages & domains, it is very unnatural to attach a property to an object at runtime and hence I believe that having an explicit explanation of what is happening and the fact that it could have been named anything else is appropriate here. 

As I say in my extended comment in the pull request (which will not be part of the notebook itself) that one should not be dynamically attaching properties in the first place.  It is a bad design of `tf.Module` that allows it (or rather mandates it) and hence at a minimum, we should explain it.

Reply:
Two things:
1. I was asking about keeping in the text:
```
# Practically this next line allows:
#
# LinearMixedEffectModel()(features_train).trainable_variables
# ==> tuple of all tf.Variables created by LinearMixedEffectModel
```

2. If you feel strongly about it, I won't insist (improving examples is great!). I don't think an example notebook is the right place to teach how Python classes work, though.